src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1159,11 +1265,36 @@ public DeployingContext(DistributedApplicationModel model, DistributedApplicatio
 
         public DistributedApplicationModel Model { get { throw null; } }
 
-        public string OutputPath { get { throw null; } }
+        public string? OutputPath { get { throw null; } }
 
         public System.IServiceProvider Services { get { throw null; } }
     }
 
+    [System.Diagnostics.DebuggerDisplay("Type = {GetType().Name,nq}")]
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIRECOMPUTE001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public sealed partial class DeploymentImageTagCallbackAnnotation : IResourceAnnotation
+    {
+        public DeploymentImageTagCallbackAnnotation(System.Func<DeploymentImageTagCallbackAnnotationContext, string> callback) { }
+
+        public DeploymentImageTagCallbackAnnotation(System.Func<DeploymentImageTagCallbackAnnotationContext, System.Threading.Tasks.Task<string>> callback) { }
+
+        public DeploymentImageTagCallbackAnnotation(System.Func<string> callback) { }
+
+        public System.Func<DeploymentImageTagCallbackAnnotationContext, System.Threading.Tasks.Task<string>> Callback { get { throw null; } }
+    }
+
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIRECOMPUTE001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public sealed partial class DeploymentImageTagCallbackAnnotationContext
+    {
+        public required System.Threading.CancellationToken CancellationToken { get { throw null; } init { } }
+
+        public required DistributedApplicationExecutionContext ExecutionContext { get { throw null; } init { } }
+
+        public required IResource Resource { get { throw null; } init { } }
+
+        public required System.IServiceProvider Services { get { throw null; } init { } }
```

Comment:
I think we can get rid of the `IServiceProvider Services` on this class. The `ExecutionContext` has an `IServiceProvider`. No reason for both of them.

Reply:
#11520